frontend/src/components/general/Sidebar/Sidebar.js

```diff
@@ -1,42 +1,97 @@
+    props.dispatch({
+      type: SET_EXPANDED,
+      payload: !isExpanded
+    })
+  }
+
   return (
-    <SidebarEdit variant='dark'>
+    <SidebarEdit>
       <NavBarTextContainer>
         <Nav.Link as={LogoDiv} to={GeneralRoutes.HOME} key={GeneralRoutes.HOME} id='logo'>
-          <p>
-            <FontAwesomeIcon style={{ marginRight: '10px' }} icon={faFire} />
-            <br />
-            Systematic Chaos
-          </p>
+          <FontAwesomeIcon icon={faFire} size={'2x'} />
+          {isExpanded && <p>Systematic Chaos</p>}
         </Nav.Link>
+
+        <div className={`d-flex w-100  ${isExpanded ? 'justify-content-end pe-3' : 'justify-content-center'}`}>
+          <FontAwesomeIcon
+            icon={isExpanded ? faAnglesLeft : faAnglesRight}
+            size={'lg'}
+            color={'white'}
+            onClick={toggleSidebar}
+            style={{ cursor: 'pointer' }}
+          />
+        </div>
+
         <NavEdit className='d-flex flex-column' id='menu-options'>
-          {link_titles.map(([to, linkTitle]) => buildNavLink(to, linkTitle[0]))}
+          {props.link_titles.map((link) => (
+            <div key={link.navigateTo} className={isExpanded ? '' : 'd-flex flex-column align-items-center'}>
+              <NavLinkStyles
+                as={Link}
+                to={link.navigateTo}
+                onClick={() => {
+                  if (link.action === 'LOGOUT') {
+                    logOut()
+                  }
+                }}
+                style={{ color: window.location.pathname === link.navigateTo ? 'var(--font-color)' : 'white' }}
+              >
+                <div style={{ width: 30, textAlign: 'center' }}>
+                  <FontAwesomeIcon icon={link.icon} />
+                </div>
+                {isExpanded && <span className={'ps-2'}>{link.name}</span>}
+              </NavLinkStyles>
 
-          <NavLinkStyles as={Link} key={'logout'} to={GeneralRoutes.HOME} onClick={() => logOut()}>
-            <span>Wyloguj</span>
-          </NavLinkStyles>
+              {link.subpages && (
+                <div
+                  className={isExpanded ? 'ms-3' : ''}
+                  style={{ borderLeft: isExpanded ? '2px solid white' : 'none' }}
+                >
+                  {link.subpages.map((sublink) => (
+                    <NavLinkStyles
+                      as={Link}
+                      key={sublink.navigateTo}
+                      to={sublink.navigateTo}
+                      className={isExpanded ? 'ps-3' : ''}
+                      style={{ color: window.location.pathname === sublink.navigateTo ? 'var(--font-color)' : 'white' }}
+                    >
+                      <div style={{ width: 30, textAlign: 'center' }}>
+                        <FontAwesomeIcon icon={sublink.icon} />
+                      </div>
+                      {isExpanded && <span className={isExpanded ? 'ps-2' : ''}>{sublink.name}</span>}
+                    </NavLinkStyles>
```

Comment:
Zastanawiam się czy nie miałoby sensu wydzielenie wspólnej funkcji na tworzenie pozycji w sidebarze z dodatkowym parametrem (boolean) który by mówił czy to link na sidebar czy nie, wydaje mi się że sporo kodu się pokrywa... ale raczej nie, chyba tak jest lepiej jak jest teraz jak przyglądam się temu dalej